instrumentation/opentelemetry/internal/metrics/metrics.go

```diff
@@ -0,0 +1,67 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/metric"
+	"log"
+)
+
+const meterName = "goagent.hypertrace.org/metrics"
+
+type systemMetrics interface {
+	getMemory() (float64, error)
+	getCPU() (float64, error)
+	getCurrentMetrics() error
+}
+
+func InitializeSystemMetrics() {
+	meterProvider := otel.GetMeterProvider()
+	meter := meterProvider.Meter(meterName)
+	err := setUpMetricRecorder(meter)
+	if err != nil {
+		log.Printf("error initializing metrics, failed to setup metric recorder: %v\n", err)
+	}
+}
+
+func setUpMetricRecorder(meter metric.Meter) error {
+	if meter == nil {
+		return fmt.Errorf("error while setting up metric recorder: meter is nil")
+	}
+	cpuSeconds, err := meter.Float64ObservableCounter("hypertrace.agent.cpu.seconds.total", metric.WithDescription("Metric to monitor total CPU seconds"))
+	if err != nil {
+		return fmt.Errorf("error while setting up cpu seconds metric counter: %v", err)
+	}
+	memory, err := meter.Float64ObservableGauge("hypertrace.agent.memory", metric.WithDescription("Metric to monitor memory usage"))
+	if err != nil {
+		return fmt.Errorf("error while setting up memory metric counter: %v", err)
+	}
+	// Register the callback function for both cpu_seconds and memory observable gauges
+	_, err = meter.RegisterCallback(
+		func(ctx context.Context, result metric.Observer) error {
+			sysMetrics := newSystemMetrics()
+			err := sysMetrics.getCurrentMetrics()
```

Comment:
now that I see these 2 together, lets merge the logic of getCurrentMetrics to the constructor and remove that function from the interface itself